src/enhancers/enhance_pickup_dropoff.rs

```diff
@@ -0,0 +1,794 @@
+            id: "default_service".to_owned(),
+            dates,
+        }])
+        .unwrap();
+        enhance_pickup_dropoff(&mut collections);
+        let vj1 = collections.vehicle_journeys.get("vj1").unwrap();
+        let stop_time = &vj1.stop_times[0];
+        assert_eq!(0, stop_time.pickup_type);
+        assert_eq!(1, stop_time.drop_off_type);
+        let stop_time = &vj1.stop_times.last().unwrap();
+        assert_eq!(1, stop_time.pickup_type);
+        assert_eq!(0, stop_time.drop_off_type);
+        let vj2 = collections.vehicle_journeys.get("vj2").unwrap();
+        let stop_time = &vj2.stop_times[0];
+        assert_eq!(0, stop_time.pickup_type);
+        assert_eq!(1, stop_time.drop_off_type);
+        let stop_time = &vj2.stop_times.last().unwrap();
+        assert_eq!(1, stop_time.pickup_type);
+        assert_eq!(0, stop_time.drop_off_type);
+    }
+
+    // Example 3
+    #[test]
+    fn stay_in_different_stop() {
+        let mut collections = Collections::default();
+        let stop_config = (
+            "block_id_1".to_string(),
+            1,
+            Time::new(10, 0, 0),
+            Time::new(11, 0, 0),
+        );
+        let next_vj_config_config = (
+            "block_id_1".to_string(),
+            2,
+            Time::new(12, 0, 0),
+            Time::new(13, 0, 0),
+        );
+        collections.vehicle_journeys = build_vehicle_journeys(stop_config, next_vj_config_config);
+        let mut dates = BTreeSet::new();
+        dates.insert(Date::from_ymd(2020, 1, 1));
+        collections.calendars = CollectionWithId::new(vec![Calendar {
+            id: "default_service".to_owned(),
+            dates,
+        }])
+        .unwrap();
+        enhance_pickup_dropoff(&mut collections);
+        let vj1 = collections.vehicle_journeys.get("vj1").unwrap();
+        let stop_time = &vj1.stop_times[0];
+        assert_eq!(0, stop_time.pickup_type);
+        assert_eq!(1, stop_time.drop_off_type);
+        let stop_time = &vj1.stop_times.last().unwrap();
+        assert_eq!(0, stop_time.pickup_type);
+        assert_eq!(0, stop_time.drop_off_type);
+        let vj2 = collections.vehicle_journeys.get("vj2").unwrap();
+        let stop_time = &vj2.stop_times[0];
+        assert_eq!(0, stop_time.pickup_type);
+        assert_eq!(0, stop_time.drop_off_type);
+        let stop_time = &vj2.stop_times.last().unwrap();
+        assert_eq!(1, stop_time.pickup_type);
+        assert_eq!(0, stop_time.drop_off_type);
+    }
+
+    // Example 3... but with route points (should not considered it a valid stay-in case)
+    // This is the same test as above, just adding an additional route-point.
+    #[test]
+    fn stay_in_different_stop_but_with_route_point() {
+        let mut collections = Collections::default();
+        let stop_config = (
+            "block_id_1".to_string(),
+            1,
+            Time::new(10, 0, 0),
+            Time::new(11, 0, 0),
+        );
+        let next_vj_config_config = (
```

Comment:
you should rename your variables (vj1_conf, and vj2_conf, or something like that)

Reply:
Indeed. It was not part of my refactoring to give such dumb names, but it seems it was me who did that in the past :exploding_head: Fixed in 406f72b.